curated_transformers/models/module.py

```diff
@@ -1,14 +1,14 @@
 from abc import abstractmethod
-from typing import Generic, List, Optional, TypeVar
+from typing import Generic, List, Optional, TypeVar, Union
 
 from torch import Tensor
 from torch.nn import Module
 
 from ..layers.attention import AttentionMask
-from .config import ConfigDataclass
+from .config import ConfigDataclass, TransformerConfig
 from .output import CacheT, CausalLMOutputWithCache, ModelOutput, ModelOutputWithCache
 
-ConfigT = TypeVar("ConfigT", bound=ConfigDataclass)
+ConfigT = TypeVar("ConfigT", bound=Union[ConfigDataclass, TransformerConfig])
```

Comment:
If we have a superclass for everything now anyway, I think we could just make `TransformerConfig` a data class. At any rate, I don't think we'd want to use the union here, because the `ConfigDataclass` bound weakens the typing. The typevar could still be useful, because it allows the non-auto classes to return the concrete type.

Reply:
I've restricted the bound to `TransformerConfig` (which is already a dataclass).